### PR TITLE
Add CI for HIP builds

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -66,7 +66,7 @@ ENV GTEST_COLOR=1
 COPY --chown=axom:axom . /home/axom/workspace
 WORKDIR /home/axom/workspace
 ENV HCC_AMDGPU_TARGET=gfx900
-RUN mkdir build && cd build && cmake -DROCM_ROOT_DIR=/opt/rocm/include -DENABLE_DEVELOPER_DEFAULTS=On -DENABLE_HIP=On ..
+RUN mkdir build && cd build && cmake -DROCM_ROOT_DIR=/opt/rocm/include -DHIP_RUNTIME_INCLUDE_DIRS="/opt/rocm/include;/opt/rocm/hip/include" -DENABLE_DEVELOPER_DEFAULTS=On -DENABLE_HIP=On ..
```
Comment (mcfadden8) [inline Dockerfile:69]: Are we not running tests with hip (or apparently nvcc)?

Comment (davidbeckingsale) [reply Dockerfile:69]: Not on Azure.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -89,6 +90,10 @@ configure_python:
     reports:
       junit: junit.xml
 
+.build_toss_3_x86_64_ib_corona_script:
```
Comment (adrienbernede) [inline .gitlab-ci.yml:93]: I guess you could just name it ".build_corona_script" if you like.

File: Dockerfile (hunk 1 above)
Comment (mcfadden8) [reply Dockerfile:69]: duh

**Review by mcfadden8 — APPROVED** [review]

File: scripts/uberenv/packages/hip/package.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            filter_file(match, substitute, *files, **kwargs)
+
+    def cmake_args(self):
+        args = [
+            '-DHIP_COMPILER=clang',
+            '-DHIP_PLATFORM=rocclr',
+            '-DHSA_PATH={0}'.format(self.spec['hsa-rocr-dev'].prefix),
+            '-DLIBROCclr_STATIC_DIR={0}/lib'.format(self.spec['rocclr'].prefix)
+        ]
+        return args
```
Comment (adrienbernede) [inline scripts/uberenv/packages/hip/package.py:57]: The hip package already exists in Spack. Why was this necessary?
Could you in particular delimit the changes with comments like `# BEGIN/END Umpire changes`?

**Review by adrienbernede — APPROVED** [review]
Oops it’s already merged...

File: scripts/uberenv/packages/hip/package.py (hunk 3 above)
Comment (davidbeckingsale) [reply scripts/uberenv/packages/hip/package.py:57]: The package in Spack is completely broken for external installs of HIP.
